src/test/java/uk/gov/hmcts/reform/sscs/validator/EnumPatternValidatorTest.java

```diff
@@ -0,0 +1,392 @@
+package uk.gov.hmcts.reform.sscs.validator;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.sscs.model.CaseCategory;
+import uk.gov.hmcts.reform.sscs.model.CaseCategoryType;
+import uk.gov.hmcts.reform.sscs.model.Dow;
+import uk.gov.hmcts.reform.sscs.model.DowUnavailabilityType;
+import uk.gov.hmcts.reform.sscs.model.HearingLocations;
+import uk.gov.hmcts.reform.sscs.model.LocationType;
+import uk.gov.hmcts.reform.sscs.model.PanelPreference;
+import uk.gov.hmcts.reform.sscs.model.PartyDetails;
+import uk.gov.hmcts.reform.sscs.model.PartyType;
+import uk.gov.hmcts.reform.sscs.model.RequirementType;
+import uk.gov.hmcts.reform.sscs.model.UnavailabilityDoW;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.sscs.exceptions.ValidationError.CATEGORY_TYPE_EMPTY;
+import static uk.gov.hmcts.reform.sscs.exceptions.ValidationError.PARTY_ROLE_EMPTY;
+import static uk.gov.hmcts.reform.sscs.exceptions.ValidationError.PARTY_TYPE_EMPTY;
+
+public class EnumPatternValidatorTest {
+
+    static Validator validator;
+
+    private static final Logger logger = LoggerFactory.getLogger(EnumPatternValidatorTest.class);
+
+    @Mock
+    private ConstraintValidatorContext constraintValidatorContext;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    @Test
+    void whenInvalidLocationIdIsNull() {
+        HearingLocations location = getHearingLocation();
+        location.setLocationId(null);
+        Set<ConstraintViolation<HearingLocations>> violations = validator.validate(location);
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
+        List<String> validationErrors = new ArrayList<>();
```

Comment:
Can't you assert directly on the violations?

Reply:
will change